src/sync/index.js

```diff
@@ -24,34 +24,246 @@ export type SyncDatabaseChangeSet = $Exact<{ [TableName<any>]: SyncTableChangeSe
 
 export type SyncLocalChanges = $Exact<{ changes: SyncDatabaseChangeSet, affectedRecords: Model[] }>
 
-export type SyncPullArgs = $Exact<{ lastSyncedAt: ?Timestamp }>
+export type SyncPullArgs = $Exact<{ lastPulledAt: ?Timestamp }>
 export type SyncPullResult = $Exact<{ changes: SyncDatabaseChangeSet, timestamp: Timestamp }>
 
-export type SyncPushArgs = $Exact<{ changes: SyncDatabaseChangeSet }>
+export type SyncPushArgs = $Exact<{ changes: SyncDatabaseChangeSet, lastPulledAt: Timestamp }>
 
 export type SyncArgs = $Exact<{
   database: Database,
   pullChanges: SyncPullArgs => Promise<SyncPullResult>,
   pushChanges: SyncPushArgs => Promise<void>,
 }>
 
+/*
+
+# Watermelon synchronization protocol
+
+To synchronize, you need to pass two functions, `pullChanges` and `pushChanges` that can talk to your
+backend.
+
+Changes are represented as an object with *raw records*. Those only use raw table and column names,
+and raw values (strings/numbers/booleans) — the same as in Schema. Deleted objects are always only
+represented by their IDs.
+
+Example:
+
+```
+{
+  projects: {
+    created: [
```

Comment:
The backend implementation we are woking on doesn't allow to detect if the record has been changed or created since the last sync. It would require to add `created_at` field to all records in collections... But the the procedure for "updated" says "if doesn't exist locally (error), create" so maybe it's not a big deal to send everything (except "deleted") in `updated`?

Reply:
oh 🤦‍♂️ don't know why I didn't realize this. will make sure this will still work correctly and make appropriate changes